### ISSUE TITLE
Walkthrough Phoenix app builds, but release application does not start

### ISSUE THREAD

**Issue by micahrye** [bitwalker/distillery#386]
### Steps to reproduce
When trying to build a release with the app would build without warning or errors but would not do anything. To sanity check I did the provided [walkthrough](https://github.com/bitwalker/distillery/blob/master/docs/Phoenix%20Walkthrough.md) (several times) and got similar results. i.e. localhost:4000 is not active. 

**reproduce** by doing project provided [walkthrough](https://github.com/bitwalker/distillery/blob/master/docs/Phoenix%20Walkthrough.md) 

### Verbose Logs

```
MIX_ENV=prod mix release --env=prod --verbose
==> Loading configuration..
==> Assembling release..
==> Building release phoenix_distillery:0.0.1 using environment prod
==> Discovered applications:
  eex-1.5.0
    from: /usr/local/Cellar/elixir/1.5.0/bin/../lib/eex
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  poison-3.1.0
    from: _build/prod/lib/poison
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  phoenix-1.3.0
    from: _build/prod/lib/phoenix
    applications:
      :kernel
      :stdlib
      :elixir
      :plug
      :poison
      :logger
      :eex
      :phoenix_pubsub
      :crypto
    includes: none

  mime-1.2.0
    from: _build/prod/lib/mime
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  plug-1.4.3
    from: _build/prod/lib/plug
    applications:
      :kernel
      :stdlib
      :elixir
      :crypto
      :logger
      :mime
    includes: none

  phoenix_html-2.10.5
    from: _build/prod/lib/phoenix_html
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :plug
    includes: none

  cowlib-1.0.2
    from: _build/prod/lib/cowlib
    applications:
      :kernel
      :stdlib
      :crypto
    includes: none

  asn1-5.0
    from: /usr/local/Cellar/erlang/20.0/lib/erlang/lib/asn1-5.0
    applications:
      :kernel
      :stdlib
    includes: none

  public_key-1.4.1
    from: /usr/local/Cellar/erlang/20.0/lib/erlang/lib/public_key-1.4.1
    applications:
      :asn1
      :crypto
      :kernel
      :stdlib
    includes: none

  ssl-8.2
    from: /usr/local/Cellar/erlang/20.0/lib/erlang/lib/ssl-8.2
    applications:
      :crypto
      :public_key
      :kernel
      :stdlib
    includes: none

  ranch-1.3.2
    from: _build/prod/lib/ranch
    applications:
      :kernel
      :stdlib
      :ssl
    includes: none

  cowboy-1.1.2
    from: _build/prod/lib/cowboy
    applications:
      :kernel
      :stdlib
      :ranch
      :cowlib
      :crypto
    includes: none

  crypto-4.0
    from: /usr/local/Cellar/erlang/20.0/lib/erlang/lib/crypto-4.0
    applications:
      :kernel
      :stdlib
    includes: none

  phoenix_pubsub-1.0.2
    from: _build/prod/lib/phoenix_pubsub
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :crypto
    includes: none

  gettext-0.14.0
    from: _build/prod/lib/gettext
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
    includes: none

  runtime_tools-1.12
    from: /usr/local/Cellar/erlang/20.0/lib/erlang/lib/runtime_tools-1.12
    applications:
      :kernel
      :stdlib
    includes: none

  logger-1.5.0
    from: /usr/local/Cellar/elixir/1.5.0/bin/../lib/logger
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  compiler-7.1
    from: /usr/local/Cellar/erlang/20.0/lib/erlang/lib/compiler-7.1
    applications:
      :kernel
      :stdlib
    includes: none

  elixir-1.5.0
    from: /usr/local/Cellar/elixir/1.5.0/bin/../lib/elixir
    applications:
      :kernel
      :stdlib
      :compiler
    includes: none

  stdlib-3.4
    from: /usr/local/Cellar/erlang/20.0/lib/erlang/lib/stdlib-3.4
    applications:
      :kernel
    includes: none

  kernel-5.3
    from: /usr/local/Cellar/erlang/20.0/lib/erlang/lib/kernel-5.3
    applications: none
    includes: none

  phoenix_distillery-0.0.1
    from: _build/prod/lib/phoenix_distillery
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :runtime_tools
      :gettext
      :phoenix_pubsub
      :cowboy
      :phoenix_html
      :phoenix
    includes: none

  iex-1.5.0
    from: /usr/local/Cellar/elixir/1.5.0/bin/../lib/iex
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  sasl-3.0.4
    from: /usr/local/Cellar/erlang/20.0/lib/erlang/lib/sasl-3.0.4
    applications:
      :kernel
      :stdlib
    includes: none

==> Generated overlay vars:
    release_name=:phoenix_distillery
    release_version="0.0.1"
    is_upgrade=false
    upgrade_from=:latest
    dev_mode=false
    include_erts=true
    include_src=false
    include_system_libs=true
    erl_opts=""
    run_erl_env=""
    erts_vsn="9.0"
    output_dir="_build/prod/rel/phoenix_distillery"
==> Copying applications to _build/prod/rel/phoenix_distillery
==> Generating nodetool
==> Generating start_erl.data
==> Generating vm.args
==> Generating sys.config from config/config.exs
==> Including ERTS 9.0 from /usr/local/Cellar/erlang/20.0/lib/erlang/erts-9.0
==> Generating boot script
==> Generating RELEASES
==> Generating start_clean.boot
==> Applying overlays
==> Applying mkdir overlay
    dst: releases/0.0.1/hooks
==> Applying mkdir overlay
    dst: releases/0.0.1/hooks/pre_configure.d
==> Applying mkdir overlay
    dst: releases/0.0.1/hooks/post_configure.d
==> Applying mkdir overlay
    dst: releases/0.0.1/hooks/pre_start.d
==> Applying mkdir overlay
    dst: releases/0.0.1/hooks/post_start.d
==> Applying mkdir overlay
    dst: releases/0.0.1/hooks/pre_stop.d
==> Applying mkdir overlay
    dst: releases/0.0.1/hooks/post_stop.d
==> Applying mkdir overlay
    dst: releases/0.0.1/hooks/pre_upgrade.d
==> Applying mkdir overlay
    dst: releases/0.0.1/hooks/post_upgrade.d
==> Applying copy overlay
    src: _build/prod/lib/distillery/priv/libexec
    dst: releases/0.0.1/libexec
==> Applying mkdir overlay
    dst: releases/0.0.1/commands
==> Packaging release..
==> Archiving phoenix_distillery-0.0.1
==> Writing tarball to _build/prod/rel/phoenix_distillery/releases/0.0.1/phoenix_distillery.tar.gz
==> Updating tarball
==> Tarball updated!
==> Release successfully built!
    You can run it in one of the following ways:
      Interactive: _build/prod/rel/phoenix_distillery/bin/phoenix_distillery console
      Foreground: _build/prod/rel/phoenix_distillery/bin/phoenix_distillery foreground
      Daemon: _build/prod/rel/phoenix_distillery/bin/phoenix_distillery start
```

### Description of issue
Running above build `PORT=4000 ./bin/phoenix_distillery foreground` as followed from "walkthrough" does not result in any output or an active phoenix application at localhost:4000. Additionally same results if I run `_build/dev/rel/phoenix_distillery/bin/phoenix_distillery foreground` etc.

- What are the expected results? Expect localhost:4000 to serve phoenix application
- What OS, Erlang/Elixir versions are you seeing this issue on?
* Mac OSX 10.11.6
* Erlang 20
* Elixir 1.5.0 and 1.6.0-rc.0 (I was using 1.6.0-rc.0 first and switch back to 1.5.0)
- If possible, also provide your `rel/config.exs`, as it is often
$ cat rel/config.exs
```
# Import all plugins from `rel/plugins`
# They can then be used by adding `plugin MyPlugin` to
# either an environment, or release definition, where
# `MyPlugin` is the name of the plugin module.
Path.join(["rel", "plugins", "*.exs"])
|> Path.wildcard()
|> Enum.map(&Code.eval_file(&1))

use Mix.Releases.Config,
    # This sets the default release built by `mix release`
    default_release: :default,
    # This sets the default environment used by `mix release`
    default_environment: Mix.env()

# For a full list of config options for both releases
# and environments, visit https://hexdocs.pm/distillery/configuration.html


# You may define one or more environments in this file,
# an environment's settings will override those of a release
# when building in that environment, this combination of release
# and environment configuration is called a profile

environment :dev do
  # If you are running Phoenix, you should make sure that
  # server: true is set and the code reloader is disabled,
  # even in dev mode.
  # It is recommended that you build with MIX_ENV=prod and pass
  # the --env flag to Distillery explicitly if you want to use
  # dev mode.
  set dev_mode: true
  set include_erts: false
  set cookie: :"t`~.Br:cU61x=r4~0Y*[<l_1GP8cbImSbSpMG:Dzc.tFGu/Xd?_E8Xx>jilo$LdS"
end

environment :prod do
  set include_erts: true
  set include_src: false
  set cookie: :"{U|~Nu_SI)e4h?X(K3FCQ<1k6v$,.c[,W*B2mxUfix,ufLr%*qCTXT=iqiXCu9LX"
end

# You may define one or more releases in this file.
# If you have not set a default release, or selected one
# when running `mix release`, the first release in the file
# will be used by default

release :phoenix_distillery do
  set version: current_version(:phoenix_distillery)
  set applications: [
    :runtime_tools
  ]
end
```

**Comment by micahrye** [comment]
Here is config/prod.exs too
```
use Mix.Config

# For production, we often load configuration from external
# sources, such as your system environment. For this reason,
# you won't find the :http configuration below, but set inside
# PhoenixDistilleryWeb.Endpoint.init/2 when load_from_system_env is
# true. Any dynamic configuration should be done there.
#
# Don't forget to configure the url host to something meaningful,
# Phoenix uses this information when generating URLs.
#
# Finally, we also include the path to a cache manifest
# containing the digested version of static files. This
# manifest is generated by the mix phx.digest task
# which you typically run after static files are built.
config :phoenix_distillery, PhoenixDistillery.Endpoint,
  http: [port: {:system, "PORT"}],
  url: [host: "localhost", port: {:system, "PORT"}], # This is critical for ensuring web-sockets properly authorize.
  cache_static_manifest: "priv/static/manifest.json",
  server: true,
  root: ".",
  version: Application.spec(:phoenix_distillery, :vsn)

# Do not print debug messages in production
config :logger, level: :info

# ## SSL Support
#
# To get SSL working, you will need to add the `https` key
# to the previous section and set your `:url` port to 443:
#
#     config :phoenix_distillery, PhoenixDistilleryWeb.Endpoint,
#       ...
#       url: [host: "example.com", port: 443],
#       https: [:inet6,
#               port: 443,
#               keyfile: System.get_env("SOME_APP_SSL_KEY_PATH"),
#               certfile: System.get_env("SOME_APP_SSL_CERT_PATH")]
#
# Where those two env variables return an absolute path to
# the key and cert in disk or a relative path inside priv,
# for example "priv/ssl/server.key".
#
# We also recommend setting `force_ssl`, ensuring no data is
# ever sent via http, always redirecting to https:
#
#     config :phoenix_distillery, PhoenixDistilleryWeb.Endpoint,
#       force_ssl: [hsts: true]
#
# Check `Plug.SSL` for all available options in `force_ssl`.

# ## Using releases
#
# If you are doing OTP releases, you need to instruct Phoenix
# to start the server for all endpoints:
#
#     config :phoenix, :serve_endpoints, true
#
# Alternatively, you can configure exactly which server to
# start per endpoint:
#
#     config :phoenix_distillery, PhoenixDistilleryWeb.Endpoint, server: true
#

# Finally import the config/prod.secret.exs
# which should be versioned separately.
import_config "prod.secret.exs"
```

**Comment by micahrye** [comment]
Also, there is ambiguity in the walkthrough document. It states `mix phoenix.new --no-ecto phoenix_distillery` which for me creates the older Phoenix strucutre: 
```
phoenix_distillery/
├── README.md
├── _build
├── brunch-config.js
├── config
├── deps
├── lib
├── mix.exs
├── mix.lock
├── node_modules
├── package.json
├── priv
├── test
└── web
```

But in section "Version 0.0.1" the command has you `cd assets` where the assets dir is part of the new Phoenix structure that gets auto generated when you use the `mix phx.new` command. 

I used the `mix phx.new --no-ecto phoenix_distillery`

**Comment by OvermindDL1** [comment]
Can you also post your Endpoint file?

**Comment by micahrye** [comment]
contents of `lib/phoenix_distillery_web/endpoint.ex` 
```
defmodule PhoenixDistilleryWeb.Endpoint do
  use Phoenix.Endpoint, otp_app: :phoenix_distillery

  socket "/socket", PhoenixDistilleryWeb.UserSocket

  # Serve at "/" the static files from "priv/static" directory.
  #
  # You should set gzip to true if you are running phoenix.digest
  # when deploying your static files in production.
  plug Plug.Static,
    at: "/", from: :phoenix_distillery, gzip: false,
    only: ~w(css fonts images js favicon.ico robots.txt)

  # Code reloading can be explicitly enabled under the
  # :code_reloader configuration of your endpoint.
  if code_reloading? do
    socket "/phoenix/live_reload/socket", Phoenix.LiveReloader.Socket
    plug Phoenix.LiveReloader
    plug Phoenix.CodeReloader
  end

  plug Plug.RequestId
  plug Plug.Logger

  plug Plug.Parsers,
    parsers: [:urlencoded, :multipart, :json],
    pass: ["*/*"],
    json_decoder: Poison

  plug Plug.MethodOverride
  plug Plug.Head

  # The session will be stored in the cookie and signed,
  # this means its contents can be read but not tampered with.
  # Set :encryption_salt if you would also like to encrypt it.
  plug Plug.Session,
    store: :cookie,
    key: "_phoenix_distillery_key",
    signing_salt: "s0QjKxj0"

  plug PhoenixDistilleryWeb.Router

  @doc """
  Callback invoked for dynamically configuring the endpoint.

  It receives the endpoint configuration and checks if
  configuration should be loaded from the system environment.
  """
  def init(_key, config) do
    if config[:load_from_system_env] do
      port = System.get_env("PORT") || raise "expected the PORT environment variable to be set"
      {:ok, Keyword.put(config, :http, [:inet6, port: port])}
    else
      {:ok, config}
    end
  end
end
```

**Comment by micahrye** [comment]
@OvermindDL1 I got it working, walking through the example a few times and banging my head I started from scratch in some different projects before coming back. I will blame my inexperience along with ambiguity in the Distillery doc and on the web :) 

The problem was that by default I used `phx.new` which creates a phoenix 1.3 app with the "new" file structure etc. As a result the the first line of the suggested edit to `config/prod.exs` was wrong. The docs have `config :phoenix_distillery, PhoenixDistillery.Endpoint,` but with the 1.3 structure it should be `PhoenixDistilleryWeb.Endpoint`. 

**It would be nice if incorrect configurations like that created some type of error or warning.**

The Distillery "[Phoenix Walkthrough](https://github.com/bitwalker/distillery/blob/master/docs/Phoenix%20Walkthrough.md)" doc  also has a decent amount of ambiguity in it as it aspects of phoenix 1.2 and 1.3 structures. 

I will clean up the doc and make a pull request.
